cmd/nodemon/nodemon.go

```diff
@@ -129,34 +132,27 @@ func run() error {
 
 	alerts := analyzer.Start(notifications)
 
-	socket, err := messaging.StartMessagingServer(nanomsgURL)
+	pubSubSocket, err := pubsub.StartPubSubMessagingServer(ctx, nanomsgPubSubURL, alerts)
+	if err != nil {
+		log.Printf("Failed to start pub sub messaging server: %v", err)
+		return err
+	}
+	defer func(socketPubSub protocol.Socket) {
+		if err := socketPubSub.Close(); err != nil {
+			log.Printf("Failed to close pubsub socket: %v", err)
+		}
+	}(pubSubSocket)
+
+	pairSocket, err := pair.StartPairMessagingServer(ctx, nanomsgPairURL, ns)
 	if err != nil {
-		log.Printf("Failed to start messaging server: %v", err)
+		log.Printf("Failed to start pair messaging server: %v", err)
 		return err
 	}
-	go func() {
-		for alert := range alerts {
-			log.Printf("Alert has been generated: %v", alert)
-
-			jsonAlert, err := json.Marshal(
-				messaging.Alert{
-					AlertDescription: alert.ShortDescription(),
-					Severity:         alert.Severity(),
-					Details:          alert.Message(),
-				})
-			if err != nil {
-				log.Printf("failed to marshal alert to json, %v", err)
-			}
-
-			message := &bytes.Buffer{}
-			message.WriteByte(byte(alert.Type()))
-			message.Write(jsonAlert)
-			err = socket.Send(message.Bytes())
-			if err != nil {
-				log.Printf("failed to send a message to socket, %v", err)
-			}
+	defer func(pairSocket protocol.Socket) {
+		if err := pairSocket.Close(); err != nil {
+			log.Printf("Failed to close pubsub socket: %v", err)
 		}
```

Comment:
I think `StartPairMessagingServer` function should be responsible to close this socket pair.